src/data_model/transformations.jl

```diff
@@ -472,3 +472,917 @@ function apply_voltage_angle_difference_bounds!(eng::Dict{String,<:Any}, vad::Re
+            # obtain impedance parameters, directly or from linecode
+            z_s, y_fr, y_to = _get_line_impedance_parameters(data_eng, line)
+
+            # remove short-circuit line and merge terminals
+            if all(iszero.(y_fr)) && all(iszero.(y_to)) && all(abs.(z_s).<=zero_series_impedance_threshold)
+                for (t_fr, t_to) in zip(line["f_connections"], line["t_connections"])
+                    _merge_terminals!(data_eng, line["f_bus"], sort([t_fr, t_to], rev=true)...)
+                end
+            # convert line to a shunt
+            else
+                shunt = _loop_line_to_shunt(data_eng, id)
+                if !haskey(data_eng, "shunt")
+                    data_eng["shunt"] = Dict{String, Any}()
+                end
+                data_eng["shunt"]["$shunt_id_prefix.$id"] = shunt
+            end
+            # delete the line now that it has been handled
+            delete!(data_eng["line"], id)
+        end
+    end
+
+    # update the 'conductor_ids' property
+    find_conductor_ids!(data_eng)
+
+    return data_eng
+end
+
+
+"Return the Kron-reduction of the specified neutral conductors of a series impedance matrix."
+function _kron_reduce_series_impedance(Z::Matrix, neutral_conductors::Vector{Int})
+    # neutral conductor idxs
+    N = neutral_conductors
+    # phase conductor idxs (complement)
+    P = setdiff(1:size(Z)[1], N)
+
+    Zkr = Z[P,P] - Z[P,N]*inv(Z[N,N])*Z[N,P]
+
+    return Zkr
+end
+
+
+"Return the Kron-reduction of the specified neutral conductors of a shunt addmittance matrix."
+function _kron_reduce_shunt_addmittance(Y::Matrix, neutral_conductors::Vector{Int})
+    # phase conductor idxs (complement)
+    P = setdiff(1:size(Y)[1], neutral_conductors)
+
+    Ykr = Y[P,P]
+
+    return Ykr
+end
+
+
+"Kron-reduce specified neutral conductors of a linecode."
+function _kron_reduce_linecode!(l, neutral_conductors::Vector{Int})
+    z_s = _kron_reduce_series_impedance(l["rs"].+im*l["xs"], neutral_conductors)
+    y_fr = _kron_reduce_shunt_addmittance(l["g_fr"].+im*l["b_fr"], neutral_conductors)
+    y_to = _kron_reduce_shunt_addmittance(l["g_to"].+im*l["b_to"], neutral_conductors)
+    l["rs"] = real.(z_s)
+    l["xs"] = imag.(z_s)
+    l["g_fr"] = real.(y_fr)
+    l["b_fr"] = imag.(y_fr)
+    l["g_to"] = real.(y_to)
+    l["b_to"] = imag.(y_to)
+end
+
+
+"Return the Kron-reduction of the specified neutral conductors of a linecode."
+_kron_reduce_linecode(l, neutral_conductors::Vector{Int}) = _kron_reduce_linecode!(deepcopy(l), neutral_conductors)
+
+
+"""
+Kron-reduce all (implied) neutral conductors of lines, switches and shunts, and remove any terminals which become unconnected. 
+A line or switch conductor is considered as a neutral conductor if it is connected between two neutral terminals. 
+A terminal is a neutral terminals if it is galvanically connected (i.e. through a line or switch) 
+to a grounded terminal, or the neutral conductor of a wye-connected component.
+"""
```

Comment:
For exported functions, we should follow the new docstring pattern where the function is given (indented) first, see, e.g. transform_data_model function as an example.